artio-codecs/src/main/java/uk/co/real_logic/artio/dictionary/generation/EncoderGenerator.java

```diff
@@ -541,6 +534,16 @@ private String generateStringSetter(
             "        %1$sLength = value.length();\n" +
             "        return this;\n" +
             "    }\n\n" +
+            "    public %3$s %1$s(final AsciiSequenceView value)\n" +
+            "    {\n" +
+            "        if (value.length() > 0)\n" +
```

Comment:
Maybe I've misunderstood something here Tom - but wasn't the problem here the buffer being null, rather than the length being 0? Also - whichever if the fields you pull out it's probably worth putting into a local variable in the generated code.

Reply:
Yes it was the buffer being null but as the buffer being null occurs at the same time as length being 0 felt the code would be more readable in this form as it is a bit more explicit about what is checking.

Will move it to a null check if you prefer

Reply:
Yeah, I think the null check would be better. People could subclass AsciiSequenceView and have a null buffer and a non-zero length for example. I think it'll optimise slightly better as well.

Reply:
Done